build-scripts/package-tests.cake

```diff
@@ -0,0 +1,382 @@
+    protected override string PackageTestBinDirectory => PackageTestDirectory + "tools/netcoreapp3.1/any/";
+    protected override string ExtensionInstallDirectory => _parameters.TestDirectory;
+
+    protected override void InstallEngineExtension(string extension)
+    {
+        _context.NuGetInstall(extension,
+            new NuGetInstallSettings() { OutputDirectory = ExtensionInstallDirectory });
+    }
+}
+
+public class ChocolateyPackageTester : NetFXPackageTester
+{
+    public ChocolateyPackageTester(BuildParameters parameters) : base(parameters) { }
+
+    protected override string PackageName => _parameters.ChocolateyPackageName;
+    protected override FilePath PackageUnderTest => _parameters.ChocolateyPackage;
+    protected override string PackageTestDirectory => _parameters.ChocolateyTestDirectory;
+    protected override string PackageTestBinDirectory => PackageTestDirectory + "tools/";
+    protected override string ExtensionInstallDirectory => _parameters.TestDirectory;
+
+    // Chocolatey packages have a different naming convention from NuGet
+    protected override string NUnitV2Driver => "nunit-extension-nunit-v2-driver";
+    protected override string NUnitProjectLoader => "nunit-extension-nunit-project-loader";
+
+    protected override void InstallEngineExtension(string extension)
+    {
+        // Install with NuGet because choco requires administrator access
+        _context.NuGetInstall(extension,
+            new NuGetInstallSettings()
+            {
+                Source = new[] { "https://chocolatey.org/api/v2/" },
+                OutputDirectory = ExtensionInstallDirectory
+            });
+    }
+}
+
+public class MsiPackageTester : NetFXPackageTester
+{
+    public MsiPackageTester(BuildParameters parameters) : base(parameters) { }
+
+    protected override string PackageName => _parameters.MsiPackageName;
+    protected override FilePath PackageUnderTest => _parameters.MsiPackage;
+    protected override string PackageTestDirectory => _parameters.MsiTestDirectory;
+    protected override string PackageTestBinDirectory => PackageTestDirectory + "NUnit.org/nunit-console/";
+    protected override string ExtensionInstallDirectory => _parameters.TestDirectory;
+
+    protected override void InstallEngineExtension(string extension)
+    {
+        _context.NuGetInstall(extension,
+            new NuGetInstallSettings() { OutputDirectory = ExtensionInstallDirectory });
+    }
+
+    protected override void CreateTestDirectory()
+    {
+        // Msiexec does not tolerate forward slashes!
+        string package = PackageUnderTest.ToString().Replace("/", "\\");
+        string testDir = PackageTestDirectory.Replace("/", "\\");
+        int rc = _context.StartProcess("msiexec", $"/a {package} TARGETDIR={testDir} /q");
+        if (rc != 0)
+            Console.WriteLine($"  ERROR: Installer returned {rc.ToString()}");
+        else
+        {
+            // Administrative install doesn't copy these files
```

Comment:
Out of interest, could you elaborate on what happens here? Does msiexec somehow not unpack all the files?